src/java_bytecode/java_utils.cpp

```diff
@@ -0,0 +1,20 @@
+/*******************************************************************\
+
+Module: JAVA Bytecode Language Conversion
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+#include <string>
+
+#include "java_utils.h"
+#include "java_types.h"
+
+constant_exprt as_number(const mp_integer value, const typet &type)
+{
+  const size_t java_int_width(type.get_unsigned_int(ID_width));
+  const std::string significant_bits(integer2string(value, 2));
+  std::string binary_width(java_int_width-significant_bits.length(), '0');
+  return constant_exprt(binary_width+significant_bits, type);
+}
```

Comment:
See my comment above: it's not clear to me why this would even be needed (I'm aware that you have only moved it around).